js/main.js

```diff
@@ -1,2 +1,24 @@
 /* global data */
 /* exported data */
+
+var $formEnterNew = document.querySelector('.new-entry');
+var $newImgURL = $formEnterNew.querySelector('img');
+
+$formEnterNew.addEventListener('input', function () {
+  event.preventDefault();
+  var $inputURL = $formEnterNew.elements.url.value;
+  $newImgURL.setAttribute('src', $inputURL);
+});
+
+$formEnterNew.addEventListener('submit', function () {
+  event.preventDefault();
+  var formObject = {};
+  formObject.title = $formEnterNew.elements.title.value;
+  formObject.url = $formEnterNew.elements.url.value;
+  formObject.notes = $formEnterNew.elements.notes.value;
+  formObject.entryId = data.nextEntryId;
+  data.nextEntryId++;
+  data.entries.push(formObject);
```

Comment:
You want to add to the beginning of the array, not the end. You should use `unshift` instead of `push`